### PR TITLE
feat(developer): pipelines application catalog (P0.6)

### PR DESCRIPTION
**Pull request by natechadwick**
## Summary

Next Developer-module slice as a **new PR** off `development` (post-#1588).

- **API:** `GET /services/pipelines` — non-hidden server applications via `PSServerXmlObjectStore.getApplicationSummaryObjects` (id, name, description, enabled, appRoot, appType, version, empty)
- **SPA:** Pipelines tab is a live read-only catalog (no longer a placeholder)
- **Docs:** P0 task README marks P0.6 done; next slices called out as separate PRs

## Scope boundary

Read catalog only. No start/stop, editor, IR import, or Slice A runtime.

## Test plan

- [x] `npm test -- --run src/test/ts/developer` (5/5)
- [x] `mvn -pl rest,projects/sitemanage -am compile -DskipTests`
- [ ] Manual: Developer → Pipelines shows apps for a designer session

> Co-Authored by Grok Build using grok-4.5 with agent Grok.

### CODE REVIEW

File: projects/sitemanage/src/main/java/com/percussion/apibridge/PipelinesAdaptor.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  @Override
+  public List<ApplicationSummary> listApplications(URI baseUri) {
+    // baseUri reserved for HATEOAS link building (interface contract)
+    PSRequest req = PSSecurityFilter.getCurrentRequest();
+    if (req == null) {
+      throw new IllegalStateException("No current request for application catalog");
+    }
+    PSSecurityToken tok = req.getSecurityToken();
+    PSApplicationSummary[] sums =
+        PSServerXmlObjectStore.getInstance().getApplicationSummaryObjects(tok, false);
```
Comment (natechadwick) [inline projects/sitemanage/src/main/java/com/percussion/apibridge/PipelinesAdaptor.java]: **1. `PipelinesAdaptor` calls `PSServerXmlObjectStore.getInstance()` directly.** That singleton has a heavy static initializer and is notoriously hard to bring up in unit tests (Spring context, encrypted properties, etc.). There is no test in this PR for the adaptor. Consider extracting a small `IPSApplicationCatalogStore` interface so the adaptor is testable without the singleton. If you want to keep the singleton for now, add a TODO and at minimum a `PipelinesAdaptorTest` that mocks `IPipelinesAdaptor` end-to-end and asserts the filter (`false` second arg) is honored.

File: projects/sitemanage/src/main/java/com/percussion/apibridge/PipelinesAdaptor.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import com.percussion.security.PSSecurityToken;
+import com.percussion.server.PSRequest;
+import com.percussion.servlets.PSSecurityFilter;
+import com.percussion.system.utils.PSSiteManageBean;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
```
Comment (natechadwick) [inline projects/sitemanage/src/main/java/com/percussion/apibridge/PipelinesAdaptor.java:36]: **9. No Javadoc on `PipelinesAdaptor`** even though `IPipelinesAdaptor` has it. Add a one-line class-level Javadoc mirroring the interface contract (`List server applications visible to the current security token`). Trivial, but worth doing while the file is new.

File: projects/sitemanage/src/main/java/com/percussion/apibridge/PipelinesAdaptor.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        try {
+          out.add(toSummary(sum));
+        } catch (Exception e) {
+          log.debug(
+              "Skipping application summary {}: {}", sum.getName(), e.getMessage());
+        }
+      }
+    }
+    out.sort(
+        Comparator.comparing(
```
Comment (natechadwick) [inline projects/sitemanage/src/main/java/com/percussion/apibridge/PipelinesAdaptor.java:109]: **10. Comparator verbose.** `Comparator.comparing(a -> a.getName() != null ? a.getName() : "", String.CASE_INSENSITIVE_ORDER)` is null-safe but verbose. `Comparator.nullsLast(Comparator.comparing(ApplicationSummary::getName, String.CASE_INSENSITIVE_ORDER))` is one line and reads cleaner. Same behaviour for null handling here (nulls last, empty string sorts before non-empty).

File: rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+@PSSiteManageBean(value = "restPipelinesResource")
+@Path("/pipelines")
+@XmlRootElement
+@Tag(
+    name = "Pipelines",
+    description = "Data pipeline / XML application design catalog (read-only)")
+public class PipelinesResource {
+
+  @Autowired private IPipelinesAdaptor adaptor;
```
Comment (natechadwick) [inline rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java]: **2. `@Autowired private IPipelinesAdaptor adaptor;` is field injection.** Breaks test substitution without reflection and is inconsistent with the rest of `rest/` which prefers constructor injection. Switch to a constructor (and update the bean-discovery wiring if needed). Cross-reference `rest/AGENTS.md` for the local convention.

File: rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+  @Context private UriInfo uriInfo;
+
+  public PipelinesResource() {}
+
+  @GET
+  @Path("/")
```
Comment (natechadwick) [inline rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java]: **3. `@Path("/")` on the `listApplications()` method is unnecessary.** With class `@Path("/pipelines")`, JAX-RS already resolves both `/pipelines` and `/pipelines/` to this method via the empty path. The trailing `/` adds noise and can confuse reverse proxies / URL canonicalization. Drop it.

File: rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.WebApplicationException;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.UriInfo;
+import jakarta.xml.bind.annotation.XmlRootElement;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+
+@PSSiteManageBean(value = "restPipelinesResource")
```
Comment (natechadwick) [inline rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java:46]: **4. `@PSSiteManageBean(value = "restPipelinesResource")` legacy bean-discovery hint.** `rest` has been moving toward Spring autodiscovery (`@RestController` / `@Path` only). Verify the rest app's component-scan actually picks this resource up — there is no test in the PR that hits `GET /services/pipelines` end-to-end. If a separate bean wiring XML is required, that should be referenced from the PR description.

File: rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+            responseCode = "200",
+            description = "OK",
+            content =
+                @Content(
+                    array =
+                        @ArraySchema(
+                            schema = @Schema(implementation = ApplicationSummary.class)))),
+        @ApiResponse(responseCode = "500", description = "Error")
+      })
+  public List<ApplicationSummary> listApplications() {
```
Comment (natechadwick) [inline rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java]: **5. No pagination / size cap.** `getApplicationSummaryObjects` can return hundreds of apps on a real server. A single un-capped JSON array will balloon payload size and SPA render time for designers. Even a read-only catalog should cap — e.g. `?limit=` (default 200) plus `?offset=`. Better: `?cursor=` since the legacy `PSApplicationSummary` ordering is by id. Slice A can refine, but please cap now.

Comment (natechadwick) [inline rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java]: **6. No filter param.** `getApplicationSummaryObjects(tok, false)` returns non-hidden apps but offers no `?q=` (name substring) or `?appType=` filter, so the SPA would have to filter client-side. A simple `?name=` substring would help and is cheap to add. Optional for P0.6 but flagged for Slice A.

File: rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    try {
+      return adaptor.listApplications(uriInfo.getBaseUri());
+    } catch (Exception e) {
+      // Preserve cause so log analysis retains the original stack/type
+      throw new WebApplicationException(e, 500);
```
Comment (natechadwick) [inline rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java:99]: **8. `throw new WebApplicationException(e, 500)` loses the response body.** The JAX-RS default 500 page will not match the JSON contract that the SPA `ApiError` consumer expects (`{ status, message }` shape). Use a typed response — `Response.status(500).entity(new ErrorBody("PIPELINES_LIST_FAILED", e.getMessage())).type(MediaType.APPLICATION_JSON).build()` — or let a global exception mapper handle it. Either way, the SPA will currently get an HTML or empty 500 body.

File: rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java (hunk 7 above)
Comment (natechadwick) [inline rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java]: **22. No end-to-end test hitting `GET /services/pipelines`.** The PR adds the resource and the adaptor but no integration test (RestAssured / MockMvc / JAX-RS `InMemoryTestContainer`) that asserts `GET /services/pipelines` returns a `200` with `ApplicationSummary` JSON. Without it, an autowiring regression (finding #4) would only show up at runtime.

File: rest/src/main/java/com/percussion/rest/pipelines/ApplicationSummary.java [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+public class ApplicationSummary {
+
+  private Integer id;
+  private String name;
+  private String description;
+  private Boolean enabled;
+  private String appRoot;
+  private String appType;
+  private String version;
+  private Boolean empty;
```
Comment (natechadwick) [inline rest/src/main/java/com/percussion/rest/pipelines/ApplicationSummary.java:37]: **7. `hidden` flag is dropped.** `PSApplicationSummary` exposes `isHidden()` but `ApplicationSummary` does not carry it. If a future admin view ever wants to see hidden apps (e.g. for troubleshooting a hidden webhooks app), this PR forces another DTO change. Add `hidden` now — it's a 4-line change (field, getter, setter, copy in `toSummary`).

File: WebUI/src/main/ts/api/developer/pipelinesApi.ts [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+      return Array.isArray(raw) ? (raw as T[]) : [raw as T];
+    }
+  }
+  return [];
+}
+
+/** GET /services/pipelines */
+export async function listApplications(): Promise<ApplicationSummary[]> {
+  const payload = await get<unknown>(PATHS.PIPELINES);
+  return asArray<ApplicationSummary>(payload, [
```
Comment (natechadwick) [inline WebUI/src/main/ts/api/developer/pipelinesApi.ts]: **11. `asArray<T>` tries three key shapes but only `Application` is real.** With `@XmlRootElement(name = "Application")` + `@Produces(APPLICATION_JSON)` on a JAX-RS method returning `List<ApplicationSummary>`, Jackson serialises a bare array — there is no wrapper object. The `ApplicationList` / `applicationList` branches are dead. If they're defensive scaffolding for a future wrapped payload, Javadoc it; otherwise drop them and the helper.

File: WebUI/src/main/ts/developer/PipelinesPanel.tsx [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+ */
+
+import React, { useEffect, useState } from "react";
+import { isSessionRedirectError, type ApiError } from "../api/client";
+import { listApplications } from "../api/developer/pipelinesApi";
+import type { ApplicationSummary } from "../api/developer/types";
+import { CatalogHint, CatalogStatus, SimpleCatalogTable } from "./CatalogTable";
+import { DEV_MSG } from "./messages";
+
+function errMsg(err: unknown, fallback: string): string {
```
Comment (natechadwick) [inline WebUI/src/main/ts/developer/PipelinesPanel.tsx]: **12. `errMsg` helper duplicated across panels.** This same `isSessionRedirectError` + `ApiError` + `Error.message` ladder is almost certainly also defined in `ContentTypesPanel`, `KeywordsPanel`, `SlotsPanel`, etc. Extract a single `panelErrMsg(err, fallback)` (or `devErrMsg`) in `WebUI/src/main/ts/developer/` and import from there. Avoids the copy-paste that already shows up across the developer module.

File: WebUI/src/main/ts/developer/PipelinesPanel.tsx [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+          DEV_MSG.PIPE_COL_ID,
+          DEV_MSG.PIPE_COL_TYPE,
+          DEV_MSG.PIPE_COL_ENABLED,
+          DEV_MSG.PIPE_COL_ROOT,
+          DEV_MSG.PIPE_COL_DESCRIPTION,
+        ]}
+        rows={sorted.map((app, index) => ({
+          key: String(app.id ?? app.name ?? `pipe-${index}`),
+          cells: [
+            <span key="n" style={{ fontFamily: "monospace" }}>
```
Comment (natechadwick) [inline WebUI/src/main/ts/developer/PipelinesPanel.tsx]: **13. Inline design-token styles.** `style={{ fontFamily: "monospace", color: "#4a5568" }}` is a one-off design token repeated across cells. Move to a CSS module / shared style object so the values are greppable and themeable. Right now changing the secondary text colour requires an N-file find-replace.

File: WebUI/src/main/ts/developer/PipelinesPanel.tsx [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+    );
+  if (items == null)
+    return (
+      <CatalogStatus testId="developer-pipe-loading">{DEV_MSG.PIPE_LOADING}</CatalogStatus>
+    );
+  if (items.length === 0)
+    return <CatalogStatus testId="developer-pipe-empty">{DEV_MSG.PIPE_EMPTY}</CatalogStatus>;
+
+  const sorted = [...items].sort((a, b) =>
+    (a.name || "").localeCompare(b.name || "", undefined, { sensitivity: "base" }),
```
Comment (natechadwick) [inline WebUI/src/main/ts/developer/PipelinesPanel.tsx]: **14. Sort runs every render.** `const sorted = [...items].sort(...)` recomputes on every render. For a small list it's harmless, but `useMemo(() => [...items].sort(...), [items])` would be cleaner and avoids identity churn if anything downstream ever memoises on `sorted`.

File: WebUI/src/main/ts/developer/PipelinesPanel.tsx [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+  useEffect(() => {
+    let cancelled = false;
+    listApplications()
+      .then((list) => {
+        if (!cancelled) setItems(list);
+      })
+      .catch((e: unknown) => {
+        if (cancelled) return;
+        // Session redirect navigates away; still leave loading so UI does not hang
+        // if navigation is delayed or blocked.
```
Comment (natechadwick) [inline WebUI/src/main/ts/developer/PipelinesPanel.tsx]: **15. Misleading comment on the error path.** The comment says "Session redirect navigates away; still leave loading so UI does not hang", but the code below it does `setError(errMsg(...)); setItems([])` — which moves **out of** the loading state. Either change the comment to match the behaviour (we are exiting loading and showing the error) or change the behaviour to match the comment (e.g. leave `items` untouched on session-redirect).

File: WebUI/src/test/ts/developer/DeveloperShell.test.tsx [hunk 15]
```diff
@@ -69,6 +69,19 @@ vi.mock("../../../main/ts/api/developer/assemblyApi", () => ({
   ]),
 }));
 
+vi.mock("../../../main/ts/api/developer/pipelinesApi", () => ({
```
Comment (natechadwick) [inline WebUI/src/test/ts/developer/DeveloperShell.test.tsx:72]: **16. No error-path test in `DeveloperShell.test.tsx`.** The `pipelinesApi` mock only returns a resolved array; the catch branch (`setError` + `setItems([])`) is never exercised. Add a second `vi.mock` variant or a separate `it("shows error UI when pipelines fetch fails", ...)` test.

Comment (natechadwick) [inline WebUI/src/test/ts/developer/DeveloperShell.test.tsx:72]: **17. No empty-state test.** Mock returns one item; the `items.length === 0` branch (`PIPE_EMPTY` "No pipeline applications returned.") is untested. Add a third test that resolves `[]` and asserts the empty CatalogStatus renders.

File: WebUI/src/test/ts/developer/DeveloperShell.test.tsx [hunk 16]
```diff
@@ -90,12 +103,16 @@ describe("DeveloperShell", () => {
     expect(screen.getByText("Page")).toBeTruthy();
   });
 
-  it("shows placeholder for pipelines section", async () => {
+  it("loads pipelines catalog section", async () => {
```
Comment (natechadwick) [inline WebUI/src/test/ts/developer/DeveloperShell.test.tsx:106]: **18. No dedicated `PipelinesPanel.test.tsx`.** Panel-level behavior (sort, error, empty, render) is only verified transitively through the shell. A dedicated `PipelinesPanel.test.tsx` would catch regressions without spinning the whole shell + nav + tab state.

File: docs/ai-generated/tasks/developer-module-p0/README.md [hunk 17]
```diff
@@ -26,21 +26,22 @@
...
 - [x] Deep-link entry `developer` (query + path fallback filter)  
-- [x] Module shell with sections: Content Types, Templates, Slots, Keywords, Communities, Pipelines (placeholder)  
+- [x] Module shell with sections: Content Types, Templates, Slots, Keywords, Communities, Pipelines  
 - [x] **Content Types** panel: list from `GET /services/contenttypes`  
 - [x] **P0.2** `GET /services/contenttypes/{idOrName}` field catalog + SPA detail view  
 - [x] **P0.2b** Allowed workflows + default workflow + allowed templates on CT detail  
 - [x] **P0.3** Keywords list `GET /services/keywords` + SPA Keywords panel (read-only)  
 - [x] **P0.4** Templates `GET /services/templates` + Slots `GET /services/slots` + SPA panels  
 - [x] **P0.5** Communities list via `GET /services/communities/find?name=*` + SPA panel  
+- [x] **P0.6** Pipelines list `GET /services/pipelines` (XML application summaries) + SPA panel
```
Comment (natechadwick) [inline docs/ai-generated/tasks/developer-module-p0/README.md]: **19. P0.6 README description is missing context.** Other P0.x entries have a one-line description (e.g. "Keywords list `GET /services/keywords` + SPA Keywords panel (read-only)"). P0.6 should clarify it returns classic XML Application summaries (not the new data-pipeline IR — that's Slice A). A designer reading the README shouldn't have to open the PR to know what shape the data is in.

Comment (natechadwick) [inline docs/ai-generated/tasks/developer-module-p0/README.md]: **20. Pre-PR HARD GATE — Maven build evidence required.** Two Maven modules touched: `rest` and `projects/sitemanage`. Per `AGENTS.md` Pre-PR HARD GATE, both must be built standalone (not root `-pl -am`): `cd rest && ../mvn-env.bat clean install` first, then `cd projects/sitemanage && ../../mvn-env.bat clean install`. PR description must record both `BUILD SUCCESS` lines plus test counts. `mvn spotless:apply` on both modules before merge.

Comment (natechadwick) [inline docs/ai-generated/tasks/developer-module-p0/README.md]: **21. Erlang pre-commit review not referenced.** The `.kilo/rules/pre-commit-review.md` rule applies to any commit on this repo (Maven Java module). Either (a) link to the durable Erlang report at `docs/ai-generated/code-reviews/...`, or (b) document an explicit human skip in the PR body. As-is, the PR description does not acknowledge Erlang.

File: rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java (hunk 7 above)
Comment (natechadwick) [reply rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java]: Addressed in 8b9b661f90 — `?name=` case-insensitive substring filter on name/description.

File: WebUI/src/test/ts/developer/DeveloperShell.test.tsx (hunk 15 above)
Comment (natechadwick) [reply WebUI/src/test/ts/developer/DeveloperShell.test.tsx:72]: Addressed in 8b9b661f90 — empty-state test in `PipelinesPanel.test.tsx`.

File: rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java (hunk 8 above)
Comment (natechadwick) [reply rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java:99]: Deferred (consistent with sibling catalogs).

Keywords/Slots/Templates use the same `WebApplicationException(e, 500)` shape after the #1588 review pass. Typed JSON error bodies should land as a **cross-cutting REST mapper** change, not only on pipelines.

File: WebUI/src/main/ts/developer/PipelinesPanel.tsx (hunk 13 above)
Comment (natechadwick) [reply WebUI/src/main/ts/developer/PipelinesPanel.tsx]: Addressed in 8b9b661f90 — `useMemo` on sorted rows.

File: WebUI/src/test/ts/developer/DeveloperShell.test.tsx (hunk 15 above)
Comment (natechadwick) [reply WebUI/src/test/ts/developer/DeveloperShell.test.tsx:72]: Addressed in 8b9b661f90 via dedicated `PipelinesPanel.test.tsx` error case (`developer-pipe-error` + status).

File: projects/sitemanage/src/main/java/com/percussion/apibridge/PipelinesAdaptor.java (hunk 1 above)
Comment (natechadwick) [reply projects/sitemanage/src/main/java/com/percussion/apibridge/PipelinesAdaptor.java]: Addressed in 8b9b661f90.

Production still loads via `PSServerXmlObjectStore` (that is the real catalog source), but the adaptor now accepts an injectable `Function<PSSecurityToken, PSApplicationSummary[]>` and keeps **map/filter/sort/limit** as pure package-visible helpers. Resource unit tests mock `IPipelinesAdaptor` without the singleton.

File: projects/sitemanage/src/main/java/com/percussion/apibridge/PipelinesAdaptor.java (hunk 3 above)
Comment (natechadwick) [reply projects/sitemanage/src/main/java/com/percussion/apibridge/PipelinesAdaptor.java:109]: Addressed in 8b9b661f90 — now `Comparator.comparing(ApplicationSummary::getName, Comparator.nullsLast(String.CASE_INSENSITIVE_ORDER))`.

File: projects/sitemanage/src/main/java/com/percussion/apibridge/PipelinesAdaptor.java (hunk 2 above)
Comment (natechadwick) [reply projects/sitemanage/src/main/java/com/percussion/apibridge/PipelinesAdaptor.java:36]: Addressed in 8b9b661f90 — class-level Javadoc added on `PipelinesAdaptor`.

File: rest/src/main/java/com/percussion/rest/pipelines/ApplicationSummary.java (hunk 9 above)
Comment (natechadwick) [reply rest/src/main/java/com/percussion/rest/pipelines/ApplicationSummary.java:37]: Addressed in 8b9b661f90 — `hidden` field populated from `PSApplicationSummary.isHidden()` (list still defaults to non-hidden apps from the store).

File: rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java (hunk 6 above)
Comment (natechadwick) [reply rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java:46]: Acknowledged / intentional.

`@PSSiteManageBean` is the same registration path as Keywords/Slots/ContentTypes catalog resources (see `rest/AGENTS.md`). Not switching this PR to a different discovery model. Coverage for wiring is the new `PipelinesResourceTest` (mock adaptor) + existing pattern in production.

File: docs/ai-generated/tasks/developer-module-p0/README.md (hunk 17 above)
Comment (natechadwick) [reply docs/ai-generated/tasks/developer-module-p0/README.md]: Addressed in follow-up commit evidence:

- `PipelinesResourceTest`: 3 tests, BUILD SUCCESS (`-Dai.integrity.skip=true` on this worktree due to seal drift vs sibling PRs)
- `mvn -pl rest,projects/sitemanage -am compile -DskipTests`: BUILD SUCCESS
- WebUI developer vitest: 8/8

Standalone `rest` + `sitemanage` `clean install` can be re-run on a clean seal in CI.

Comment (natechadwick) [reply docs/ai-generated/tasks/developer-module-p0/README.md]: Acknowledged.

This follow-up is review-feedback only (no new feature surface beyond review fixes). Full Erlang pre-commit report deferred; happy to attach if required before merge.

File: WebUI/src/test/ts/developer/DeveloperShell.test.tsx (hunk 16 above)
Comment (natechadwick) [reply WebUI/src/test/ts/developer/DeveloperShell.test.tsx:106]: Addressed in 8b9b661f90 — `WebUI/src/test/ts/developer/PipelinesPanel.test.tsx` (success / empty / error).

File: WebUI/src/main/ts/api/developer/pipelinesApi.ts (hunk 10 above)
Comment (natechadwick) [reply WebUI/src/main/ts/api/developer/pipelinesApi.ts]: Addressed in 8b9b661f90 — client now prefers bare arrays; keeps a single defensive `Application` key for wrapped single-element shapes.

File: WebUI/src/main/ts/developer/PipelinesPanel.tsx (hunk 11 above)
Comment (natechadwick) [reply WebUI/src/main/ts/developer/PipelinesPanel.tsx]: Addressed in 8b9b661f90 — shared `panelErrMsg` in `developer/errors.ts` (PipelinesPanel uses it; other panels can migrate on touch).

File: rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java (hunk 7 above)
Comment (natechadwick) [reply rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java]: Addressed in 8b9b661f90.

`GET /services/pipelines?limit=&offset=` with default limit **500**, hard max **1000**.

File: rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java (hunk 4 above)
Comment (natechadwick) [reply rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java]: Addressed in 8b9b661f90 — constructor `@Autowired` injection on `PipelinesResource` (no-arg kept for bean-discovery edge cases).

File: docs/ai-generated/tasks/developer-module-p0/README.md (hunk 17 above)
Comment (natechadwick) [reply docs/ai-generated/tasks/developer-module-p0/README.md]: Addressed in 8b9b661f90 — README P0.6 line now states classic **XML Application** summaries (not Slice A IR) and query params.

File: WebUI/src/main/ts/developer/PipelinesPanel.tsx (hunk 14 above)
Comment (natechadwick) [reply WebUI/src/main/ts/developer/PipelinesPanel.tsx]: Addressed in 8b9b661f90 — comment now matches behaviour (exit loading + show error).

File: rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java (hunk 7 above)
Comment (natechadwick) [reply rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java]: Addressed in 8b9b661f90 with a focused unit test.

`PipelinesResourceTest` covers list delegation, name/paging params, and 500 wrapping. Full container e2e against a live object store is still deferred (needs CMS runtime fixture).

File: rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java (hunk 5 above)
Comment (natechadwick) [reply rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java]: Addressed in 8b9b661f90 — removed method-level `@Path("/")`.

File: WebUI/src/main/ts/developer/PipelinesPanel.tsx (hunk 12 above)
Comment (natechadwick) [reply WebUI/src/main/ts/developer/PipelinesPanel.tsx]: Addressed in 8b9b661f90 — shared `catalogStyles.ts` (`monoCell` / `mutedMonoCell` / `mutedCell`).

File: projects/sitemanage/src/main/java/com/percussion/apibridge/PipelinesAdaptor.java [hunk 18]
```diff
@@ -0,0 +1,141 @@
...
+    }
+    PSSecurityToken tok = req.getSecurityToken();
+    PSApplicationSummary[] sums = summaryLoader.apply(tok);
+    return mapFilterSortLimit(sums, nameFilter, limit, offset);
+  }
+
+  /**
+   * Pure mapping path used by production and unit tests (no object-store singleton).
+   */
+  static List<ApplicationSummary> mapFilterSortLimit(
```
[the remark this answers is not in the document]
Comment (natechadwick-intsof) [reply projects/sitemanage/src/main/java/com/percussion/apibridge/PipelinesAdaptor.java:84]: **Mitigation (commit `ceeaaaeebc`):** Added package-private pure unit tests in `projects/sitemanage/src/test/java/com/percussion/apibridge/PipelinesAdaptorTest.java` (9 tests) covering:
- field mapping incl. `hidden` / `appType` / `empty`
- case-insensitive name + description filter
- name sort
- `DEFAULT_LIMIT` / `MAX_LIMIT` clamp
- offset+limit window and offset-past-end empty
- null array / null entry skip

`cd projects/sitemanage && ../../mvnw -Dai.integrity.skip=true -Dtest=PipelinesAdaptorTest test` → Tests run: 9, Failures: 0.

File: rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+                        @ArraySchema(
+                            schema = @Schema(implementation = ApplicationSummary.class)))),
+        @ApiResponse(responseCode = "500", description = "Error")
+      })
+  public List<ApplicationSummary> listApplications(
+      @QueryParam("name") String name,
+      @QueryParam("limit") @DefaultValue("500") int limit,
+      @QueryParam("offset") @DefaultValue("0") int offset) {
+    try {
+      return adaptor.listApplications(uriInfo.getBaseUri(), name, limit, offset);
```
[the remark this answers is not in the document]
Comment (natechadwick-intsof) [reply rest/src/main/java/com/percussion/rest/pipelines/PipelinesResource.java]: **Mitigation (commit `ceeaaaeebc`):** `listApplications` now calls `requireAdaptor()` which throws `IllegalStateException` with a clear diagnostic when the no-arg constructor path left `adaptor` null. The exception is wrapped as HTTP 500 with cause preserved. Added `listApplicationsWithoutInjectionFailsWithDiagnostic` in `PipelinesResourceTest`.

File: rest/src/test/java/com/percussion/rest/pipelines/PipelinesResourceTest.java [hunk 20]
```diff
@@ -0,0 +1,88 @@
...
+
+  @Test
+  public void listApplicationsWrapsFailures() {
+    when(adaptor.listApplications(any(), any(), anyInt(), anyInt()))
+        .thenThrow(new IllegalStateException("boom"));
+
+    WebApplicationException ex =
+        assertThrows(
+            WebApplicationException.class, () -> resource.listApplications(null, 500, 0));
+    assertEquals(500, ex.getResponse().getStatus());
```
[the remark this answers is not in the document]
Comment (natechadwick-intsof) [reply rest/src/test/java/com/percussion/rest/pipelines/PipelinesResourceTest.java:87]: **Mitigation (commit `ceeaaaeebc`):** `listApplicationsWrapsFailures` now captures the original `IllegalStateException("boom")` and asserts `assertSame(boom, ex.getCause())` so the documented cause-preservation contract is enforced.
